src/lib/substituteResponsiveAtRules.js

```diff
@@ -29,7 +29,18 @@ export default function(config) {
           return cloned
         })
       )
-      css.append(mediaQuery)
+
+      let includesScreenUtilitiesExplicitly = false
+      css.walkAtRules('tailwind', atRule => {
+        if (atRule.params === 'screen-utilities') {
+          includesScreenUtilitiesExplicitly = true
+          atRule.replaceWith(mediaQuery)
```

Comment:
Right now you're doing this inside the loop that writes each media query, so the `screen-utilities` marker is going to get removed on the very first run of the loop and then the rest of the media queries will be added to the end of the stylesheet again.

You'll want to do the replacement at the very end, outside of the loop, and instead of writing those media queries in the loop, you'll just want to build them up and put them all into an array so you can replace the marker with all of the media queries at once later.

Reply:
Oooops, you're right. I'll fix it as soon we have a good test.